### PR TITLE
Search events

### PR DESCRIPTION
**Pull request by DaveCSW**

### CODE REVIEW

File: backend/auth/auth.go [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+	"github.com/golang-jwt/jwt/v5"
+)
+
+func IsUCSDEmail(email string) bool {
+	// Check if the email has the "ucsd.edu" domain
+	return strings.HasSuffix(email, "ucsd.edu")
+}
+
+func VerifyJWTToken(c *gin.Context, token string) {
+	jwksURL := "https://www.googleapis.com/oauth2/v3/certs"
```
Comment (gurnoor6) [inline backend/auth/auth.go:18]: Let's add this to a separate file `constants.go`

File: backend/events/tests/images/1709950517375854600.jpg2626803360 [hunk 2]
```diff
@@ -0,0 +1 @@
+test image content
```
Comment (gurnoor6) [inline backend/events/tests/images/1709950517375854600.jpg2626803360:1]: Let's delete these files. They are not required in the repository

File: backend/login/login.go [hunk 3]
```diff
@@ -1,74 +1,97 @@
...
+	Picture string `json:"picture"`
 }
 
-const oauthStateString = "random"
+func getConfig() Config {
+	return Config{
+		ClientID:        os.Getenv("GOOGLE_CLIENT_ID"),
+		ClientSecret:    os.Getenv("GOOGLE_CLIENT_SECRET"),
+		AuthURL:         "https://accounts.google.com/o/oauth2/v2/auth",
+		TokenURL:        "https://oauth2.googleapis.com/token",
```
Comment (gurnoor6) [inline backend/login/login.go:39]: Let's add these also to `constants.go`

File: backend/login/login.go [hunk 4]
```diff
@@ -1,74 +1,97 @@
...
+		RedirectURL:     os.Getenv("REDIRECT_URL"),
+		ClientURL:       os.Getenv("CLIENT_URL"),
+		TokenSecret:     os.Getenv("TOKEN_SECRET"),
+		TokenExpiration: 36000,
+		PostURL:         "https://jsonplaceholder.typicode.com/posts",
```
Comment (gurnoor6) [inline backend/login/login.go:44]: Let's add this also to constants.go

File: frontend/src/Login.jsx [hunk 5]
```diff
@@ -12,21 +12,50 @@ const Login = () => {
     const [showErrorModal, setShowErrorModal] = useState(false);
     const [errorMessage, setErrorMessage] = useState("");
 
+    const config = {
+        clientID: process.env.REACT_APP_GOOGLE_CLIENT_ID,
+        authURL: 'https://accounts.google.com/o/oauth2/v2/auth',
+        tokenURL: 'https://oauth2.googleapis.com/token',
```
Comment (gurnoor6) [inline frontend/src/Login.jsx:18]: Let's add these constants to a separate file

**Review by gurnoor6 — APPROVED** [review]

**Review by sudolucy — APPROVED** [review]

**Pull request merged**